### ISSUE TITLE
Test cases for dssp are failing

### ISSUE THREAD

**Issue by peastman** [mdtraj/mdtraj#1070]
I get two test failures when I run test_dssp.py.  I need to figure out what's going on with that before I can start cleaning up the SSE code for them.

I notice that AppVeyor shows the test as passing, but only because mkdssp isn't installed, which means it skips the relevant tests.  And travis never even gets as far as running tests.  It fails with the message

```
Error:  Package missing in current linux-64 channels: 
  - shiftx2
```

In spite of that it still shows the build passing on the front page, which seems like a problem...

**Comment by rmcgibbo** [comment]
Yes, this seems problematic. I think Matt was just working on the shiftx2
recipe on omnia, so this might be recent, or maybe not. Have you tried
running the tests locally though?

On Mon, Mar 14, 2016 at 6:47 PM, peastman <notifications@github.com> wrote:

> I get two test failures when I run test_dssp.py. I need to figure out
> what's going on with that before I can start cleaning up the SSE code for
> them.
>
> I notice that AppVeyor shows the test as passing, but only because mkdssp
> isn't installed, which means it skips the relevant tests. And travis never
> even gets as far as running tests. It fails with the message
>
> Error:  Package missing in current linux-64 channels:
>   - shiftx2
>
> In spite of that it still shows the build passing on the front page, which
> seems like a problem...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mdtraj/mdtraj/issues/1070>.
>



-- 
-Robert

**Comment by peastman** [comment]
Yes.  I get two failures in test_dssp.py.

**Comment by rmcgibbo** [comment]
Got it. Probably something is wrong then. I would put it at 70/30 that the
test case itself is buggy, but it could also be the implementation.

On Mon, Mar 14, 2016 at 7:11 PM, peastman <notifications@github.com> wrote:

> Yes. I get two failures in test_dssp.py.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mdtraj/mdtraj/issues/1070#issuecomment-196559085>.
>



-- 
-Robert

**Comment by mpharrigan** [comment]
I wad playing with it, but none of those changes have made it into master yet

**Comment by peastman** [comment]
It looks to me like this is a real issue in the code.  The test in question calls `compute_dssp()`, then executes the `mkdssp` application, reads in the results, and compares them.  On just a small number of residues, mdtraj gives different assignments:

```
 32: 'T' 'T'
 33: 'T' 'T'
 34: 'B' 'E' <-
 35: 'B' 'E' <-
 36: 'T' 'T'
 37: 'T' 'T'
 38: 'S' 'S'
 39: ' ' ' '
 40: 'B' ' ' <-
 41: ' ' ' '
 42: ' ' ' '
```

So the question is why they're disagreeing.

**Comment by peastman** [comment]
I finally tracked down the problem and fixed it: https://github.com/peastman/mdtraj/commit/af158d6aa5409fa21b9da3c541243701918c6b30.  The error was in the code that joins multiple bridges into a beta sheet with a bulge in the middle.  It wasn't checking that the ordering of the residues was actually consistent with the type of beta sheet (parallel or antiparallel) they were being merged into.  All tests now pass and give results identical to `mkdssp`.

**Comment by rmcgibbo** [comment]
Excellent!

On Thu, Mar 17, 2016 at 3:33 PM, peastman <notifications@github.com> wrote:

> I finally tracked down the problem and fixed it: peastman@af158d6
> <https://github.com/peastman/mdtraj/commit/af158d6aa5409fa21b9da3c541243701918c6b30>.
> The error was in the code that joins multiple bridges into a beta sheet
> with a bulge in the middle. It wasn't checking that the ordering of the
> residues was actually consistent with the type of beta sheet (parallel or
> antiparallel) they were being merged into. All tests now pass and give
> results identical to mkdssp.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/mdtraj/mdtraj/issues/1070#issuecomment-198048193>
>



-- 
-Robert

**Comment by mpharrigan** [comment]
Fixed in #1077
